src/hooks/useUpdateUrlSortParamOnTableChange.ts

```diff
@@ -0,0 +1,37 @@
+import { SortDirection } from "gql/generated/types";
+import { PatchTasksQueryParams, TableOnChange } from "types/task";
+import { useHistory, useLocation } from "react-router-dom";
+import { parseQueryString, stringifyQuery } from "utils";
+
+export const useUpdateUrlSortParamOnTableChange = <T extends unknown>() => {
+  const { replace } = useHistory();
+  const { search, pathname } = useLocation();
+
+  const tableChangeHandler: TableOnChange<T> = (
+    ...[, , { order, columnKey }]
+  ) => {
+    // order is undefined when the column sorter is unselected (which occurs after being clicked three times)
+    // when order is undefined, sort should be reset; therefore removed from the url
+    if (!order) {
+      replace(pathname);
+    }
+
+    const queryParams = parseQueryString(search);
+
+    const nextQueryParams = {
+      ...queryParams,
+      [PatchTasksQueryParams.SortDir]:
```

Comment:
I think it makes sense to create an enum specific to Tables instead of using what exists in PatchTaskQueryParams

Reply:
i agree that it makes sense although I do not think it's necessary. i left it as `PatchTasksQueryParams` because the sortDir and sortBy params should all be the same. however making an enum specifically for the url query params will possibly be more explicit.